From stable-bounces@linux.kernel.org  Thu Jul  6 13:00:36 2006
Date: Thu, 06 Jul 2006 14:26:45 -0400
From: Michael Krufky <mkrufky@linuxtv.org>
To: stable@kernel.org
Cc: v4l-dvb maintainer list <v4l-dvb-maintainer@linuxtv.org>
Subject: dvb-bt8xx: fix frontend detection for DViCO FusionHDTV DVB-T Lite rev 1.2

From: Michael Krufky <mkrufky@linuxtv.org>

This patch adds support for the new revision of the DViCO
FusionHDTV DVB-T Lite, based on the zl10353 demod instead
of mt352.

Both mt352 and zl10353 revisions of this card have the
same PCI subsystem ID.

Signed-off-by: Michael Krufky <mkrufky@linuxtv.org>
Acked-by: Chris Pascoe <c.pascoe@itee.uq.edu.au>
Acked-by: Manu Abraham <manu@linuxtv.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 drivers/media/dvb/bt8xx/dvb-bt8xx.c |   10 ++++++++++
 drivers/media/dvb/bt8xx/dvb-bt8xx.h |    1 +
 2 files changed, 11 insertions(+)

--- linux-2.6.17.3.orig/drivers/media/dvb/bt8xx/dvb-bt8xx.c
+++ linux-2.6.17.3/drivers/media/dvb/bt8xx/dvb-bt8xx.c
@@ -184,6 +184,11 @@ static struct mt352_config thomson_dtt75
 	.pll_set = thomson_dtt7579_pll_set,
 };
 
+static struct zl10353_config thomson_dtt7579_zl10353_config = {
+	.demod_address = 0x0f,
+	.pll_set = thomson_dtt7579_pll_set,
+};
+
 static int cx24108_pll_set(struct dvb_frontend* fe, struct dvb_frontend_parameters* params)
 {
 	u32 freq = params->frequency;
@@ -617,6 +622,11 @@ static void frontend_init(struct dvb_bt8
 	switch(type) {
 	case BTTV_BOARD_DVICO_DVBT_LITE:
 		card->fe = mt352_attach(&thomson_dtt7579_config, card->i2c_adapter);
+
+		if (card->fe == NULL)
+			card->fe = zl10353_attach(&thomson_dtt7579_zl10353_config,
+						  card->i2c_adapter);
+
 		if (card->fe != NULL) {
 			card->fe->ops->info.frequency_min = 174000000;
 			card->fe->ops->info.frequency_max = 862000000;
--- linux-2.6.17.3.orig/drivers/media/dvb/bt8xx/dvb-bt8xx.h
+++ linux-2.6.17.3/drivers/media/dvb/bt8xx/dvb-bt8xx.h
@@ -37,6 +37,7 @@
 #include "cx24110.h"
 #include "or51211.h"
 #include "lgdt330x.h"
+#include "zl10353.h"
 
 struct dvb_bt8xx_card {
 	struct mutex lock;
